### ISSUE TITLE
Passengers not removed if failed flight mission is recovered

### ISSUE THREAD

**Issue by caerfinon** [caerfinon/KerbinSideRemasteredGAP#15]
On contract failure the passengers are not removed from game if player recovers the flight and the passengers are not dead.  
- Need to add a remove kerbal behaviours on contract failure

**Comment by caerfinon** [comment]
Code to fix tourists on fail
```
// Eliminate craft on contract failure to prevent tourist spam in Astronaut Complex	
	BEHAVIOUR
	{
		name = DestroyVessel
		type = DestroyVessel
		onState = CONTRACT_FAILED
		vessel = @/craft
	}
//----------------------------------------------------------------------
//Craft definition
	PARAMETER
	{
		name = VesselParameterGroup
		type = VesselParameterGroup
		title = Your aircraft must
		
		define = @/craft
		// Do not dissassociate on failure this prevents vessel destruction behaviour
		dissassociateVesselsOnContractFailure = false


//----------------------------------------------------------------------
		DIALOG_BOX
		{
			condition = CONTRACT_FAILED
			
			position = CENTER
			title = <b>Sometimes Things Go Wrong</b>
			titleColor = #BADA55
			
			TEXT
			{
				text = <color=red><b>KSC ATC:</b> Trans One Zero Five One, Are you there? please respond. Trans One Zero Five One, we've lost you on radar, do you copy? Are you declaring an emergency?.... and you were doing so well.</color>    
			}
			
			IMAGE
			{
				url = ContractPacks/KerbinSideRemasteredGAP/Assets/Characters/@/KSC_ATC
				characterName = Air Traffic Controller
			}
			
		}		
```

Everyone on vessel is killed. leaves you with options to revert flight or go back to space center. If your game settings don't allow reverting flights then everyone is dead.

**Comment by caerfinon** [comment]
Solution has been play-tested in Dev. Upgrading flights with new code for next release

**Comment by caerfinon** [comment]
resolved by version 1.13
